specutils/analysis/flux.py

```diff
@@ -116,3 +118,57 @@ def _compute_equivalent_width(spectrum, continuum=1, regions=None):
     ew =  dx - (line_flux / continuum)
 
     return ew.to(calc_spectrum.spectral_axis.unit)
+
+def is_continuum_near_zero(spectrum, eps=0.01):
+    """
+    Determine if the baseline continuum of the spectrum is near zero.
+
+    The value is scaled to match the sigma/standard deviation parameter of a
+    standard Gaussian profile. This will be calculated over the regions, if
+    they are specified.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object over which the width will be calculated.
+
+    eps: float
+        The tolerance on the quantification to confirm the continuum is
+        near zero.
+
+    Returns
+    -------
+    is_near_zero: bool
+        A True if the continuum of the spectrum is with eps, False otherwise.
+
+    Notes
+    -----
+    The spectrum should be continuum subtracted before being passed to this
+    function.
+    """
+
+    # If the eps has units then the assumption is that we want
+    # to compare the median of the flux regardless of the
+    # existence of the uncertainty.
+    if hasattr(eps, 'unit'):
+        return np.median(spectrum.flux) < eps
+
+    # If eps does not have a unit, ie it is not a quantity, then
+    # we are going to calculate based on the S/N if the uncertainty
+    # exists.
+    if hasattr(spectrum, 'uncertainty') and spectrum.uncertainty:
+        return np.median(spectrum.flux / spectrum.uncertainty.quantity) < eps
+    else:
+        raise Exception('Spectrum flux has units, eps does not, either include uncertainty or use units on eps')
```

Comment:
I think this `else` case should instead use the  median absolute deviation (in the "scaled to be like sigma" form - https://docs.astropy.org/en/stable/api/astropy.stats.mad_std.html ) as the "uncertainty" estimate, rather than raising an error.  Then the decorator can default to using that case with an `eps` of .01 or something like that, and we can count on this function always running on a spectrum.

(This is what makes it a heuristic for continuum subtraction - not all spectra that have their continuum subtracted really will have this property, but it's a reasonable guess for many spectra.)

Reply:
Right, but, this was my question in https://github.com/astropy/specutils/issues/535#issuecomment-542339293.

Reply:
Oops, sorry - to finish out that conversation here, I don't really understand your point there.  In the case of your example linked there, that looks to have done exactly what was expected: the fluxes are close together, and that closeness indicates the "uncertainty per pixel" is probably low, so therefore the fact that the average offset from zero is larger than the scatter is *exactly* the sign of "continuum not subtracted" that we are looking for.